Services/UserCourseStatusHistorizing/classes/class.ilUserCourseStatusHistorizingHelper.php

```diff
@@ -204,18 +204,25 @@ public function setIndividualStartAndEnd($user_id, $course_id, &$payload) {
 		$case_id = array( 'usr_id'	 =>	(int)$user_id
 						, 'crs_id'	 =>	(int)$course_id
 						);
-		
+
 		if (!ilUserCourseStatusHistorizing::caseExists($case_id)) {
 			$payload["begin_date"] = date("Y-m-d");
 			return;
 		}
-		
+
+		$states_checks = array("kostenpflichtig storniert", "kostenfrei storniert");
+		if (ilUserCourseStatusHistorizing::caseExists($case_id)
+			&& in_array($payload["booking_status"], $states_checks)
+			&& $payload["event"] == "addParticipant")
+		{
+			$payload["begin_date"] = date("Y-m-d");
+		}
+
```

Comment:
in payload["booking_stgatus"] steht der aktuelle status drin. sprich ist der benutzer storniert dann steht "kostenfrei storniert"